### ISSUE TITLE
lang folder moved in laravel 9.x

### ISSUE THREAD

**Issue by fjeansson** [amiranagram/localizator#41]
the lang folder is no longer under resources, so the following line in helpers.php is incorrect for laravel 9.x installs.

return resource_path('lang'.($path !== '' ? DIRECTORY_SEPARATOR.$path : ''));

**Comment by zamon** [comment]
same with me, error creating directory when using laravel 9.x

**Comment by amiranagram** [comment]
If you noticed, there's a condition: `! function_exists('lang_path')` so the function won't be registered in Laravel 9. Not sure how you have this problem, since I'm using this package on L9 and older versions.

**Comment by amiranagram** [comment]
FYI, found this in Laravel's codebase:

```php
$this->useLangPath(value(function () {
    if (is_dir($directory = $this->resourcePath('lang'))) {
        return $directory;
    }

    return $this->basePath('lang');
}));
```

So, make sure `lang` folder doesn't exist in resources otherwise that directory will be registered as the default lang path.

**Comment by pohjolal** [comment]
> If you noticed, there's a condition: `! function_exists('lang_path')` so the function won't be registered in Laravel 9. Not sure how you have this problem, since I'm using this package on L9 and older versions.

This is still a problem in Laravel 9. It is a problem because the lang_path-function that you declare in helpers.php actually overrides Laravels own lang_path-function. Your packages helpers.php is run before Laravels [Illuminate/Foundation/helpers.php](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Foundation/helpers.php#L508) so your version of the function is used. There is a test for function_exists('lang_path') also in Laravels helpers.php.

If you want to be backwards compatible and support Laravel before lang_path then you might have to detect this at runtime.

**Comment by amiranagram** [comment]
Fixed in 0.11.1

**Comment by hirschadway** [comment]
This piece of code from the application.php shows that lang_path() function first looks for the old path( resource/lang), if for any reason it is this directory, it returns the same one, if not, it returns the new path (correct path).
```php
$this->useLangPath(value(function () {
            return is_dir($directory = $this->resourcePath('lang'))
                        ? $directory
                        : $this->basePath('lang');
```
So for the lang_path() to work properly. Delete or move the old directory created for any reason
